src/controllers/runtime/RuntimeMetadataController.ts

```diff
@@ -65,4 +73,64 @@ export default class RuntimeMetadataController extends AbstractController<Runtim
 			metadataOpts: { registry, version: metadata.version },
 		});
 	};
+
+	/**
+	 * Get the chain's metadata at a specific version in a decoded, JSON format.
+	 *
+	 * @param _req Express Request
+	 * @param res Express Response
+	 */
+	private getMetadataVersioned: RequestHandler = async (
+		{ params: { metadataVersion }, query: { at } },
+		res,
+	): Promise<void> => {
+		const hash = await this.getHashFromAt(at);
+
+		let api;
+		if (at) {
+			api = await this.api.at(hash);
+		} else {
+			api = this.api;
+		}
+
+		// Validation of the `metadataVersion` path parameter.
+		const metadataV = metadataVersion.slice(1);
+		const version = this.parseNumberOrThrow(
+			metadataV,
+			`Version ${metadataV.toString()} of metadata provided is not a number.`,
+		);
+
+		const regExPattern = new RegExp('^[vV][0-9]+$');
+		if (regExPattern.test(metadataVersion) === false) {
+			throw new Error(
+				`${metadataVersion} input is not of the expected 'vX' format, where 'X' represents the version number (examples: 'v14', 'v15').`,
+			);
+		}
+
+		const availableVersions = (await api.call.metadata.metadataVersions()).toHuman() as string[];
```

Comment:
I think `toJSON()` might be more fitting here. Generally speaking I think `toHuman` is good for user facing data that is meant to be displayed, such as UI's.

Reply:
That's awesome! Thank you so much for this one! 

Changed it and added the `as unknown as Vec<u32>` since this reflects the actual type which is returned from `await api.call.metadata.metadataVersions()`. If inspected with `.toRawType()`, it is a `Vec<u32>`.